### ISSUE TITLE
ExtendedItemSet.Remove(item) error

### ISSUE THREAD

**Issue by Artoymyp** [xBimTeam/XbimEssentials#129]
Hello.
I have created a code to remove a related object from some relationship (please see the test code below).
Resulting RelatedObject collection unexpectedly contains <null> for each removed item.
Such null-entries are also saved to an ifc-file as '$'.
Is there a way to get rid of these null-entries?

PS: Changing the type of the 'rel' variable to concrete Ifc2x3 type solves the issue. Please note that it is unacceptable for me, because the code should be schema-independent.
PSS: I'm using master branch.

```
	// arrange
	IfcStore model = IfcStore.Create(IfcSchemaVersion.Ifc2X3, XbimStoreType.InMemoryModel);
	model.BeginTransaction();

	IfcWall wall1 = model.Instances.New<IfcWall>();
	IfcWall wall2 = model.Instances.New<IfcWall>();

	IfcPropertySet propertySet = model.Instances.New<IfcPropertySet>();
	propertySet.HasProperties.Add(model.Instances.New<IfcPropertySingleValue>());

	// Changing the type of the rel variable from interface to IfcRelDefinesByProperties solves the issue.
	IIfcRelDefinesByProperties rel = model.Instances.New<IfcRelDefinesByProperties>();

	rel.RelatingPropertyDefinition = propertySet;
	rel.RelatedObjects.Add(wall1);
	rel.RelatedObjects.Add(wall2);

	// act
	rel.RelatedObjects.Remove(wall1);

	// assert failed. Expected: 1, actual: 2 (null, wall2).
	Assert.AreEqual(1, rel.RelatedObjects.Count);

**Comment by martin1cerny** [comment]
This looks like a bug in the collection used for schema independent data modifications. I'll have a look.
Martin

**Comment by martin1cerny** [comment]
Bug is fixed in 4.0.30-V005 Nuget package from `develop` feed.
